starboard/evergreen/shared/launcher.py

```diff
@@ -280,23 +280,24 @@ def _StageTargetsAndContentsRdk(self):
     content_subdir = os.path.join('usr', 'share', 'cobalt')
 
     # The rdk loader always loads libcobalt.so. Hard code target_name to match.
-    target_name = 'cobalt'
+    rdk_target_name = 'cobalt'
 
     # Copy target content and binary.
     target_install_path = os.path.join(self.out_directory, 'install')
     target_staging_dir = os.path.join(self.staging_directory, 'content', 'app',
-                                      target_name)
+                                      'cobalt')
```

Comment:
Are you sure this should be changed to `'cobalt'`? The change this makes is that all targets will share the staging directory, thereby mixing `data_deps` files. What happens if you don't change this?

Reply:
You are right. It should be rdk_target_name as well, because it is hard coded to 'cobalt' for the same reason we use rdk_target_name. In the future if we had the flexibility on the file name/path, it better be looked at again.

Reply:
I agree with your concern. And I updated this to keep the self.target_name as the destination folder name, e.g. it will be "base_unittests" when the target is base_unittests. 
This folder name will be copied to the partition /data on the RDK device. Only the symlink will be called 'cobalt' to make the loader happy.

And you can see that this change also affects how the symlink change (https://github.com/youtube/cobalt/pull/1697) merged earlier. To keep related stuff together in one CL, I have the it reverted as PR1713. Let's address that target name issue in this PR.

Reply:
Thanks. Let's just drop the other revert PR.
This one fixes the feature and takes care of the differences in merging, as a good patch should do.
Thanks!